x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/EsqlMediaTypeParser.java

```diff
@@ -53,15 +58,22 @@ private static MediaType mediaTypeFromParams(RestRequest request) {
         return MEDIA_TYPE_REGISTRY.queryParamToMediaType(request.param(URL_PARAM_FORMAT));
     }
 
-    private static MediaType validateColumnarRequest(boolean requestIsColumnar, MediaType fromMediaType, RestRequest request) {
+    private static void validateColumnarRequest(boolean requestIsColumnar, MediaType fromMediaType) {
         if (requestIsColumnar && fromMediaType instanceof TextFormat) {
             throw new IllegalArgumentException(
                 "Invalid use of [columnar] argument: cannot be used in combination with "
                     + Arrays.stream(TextFormat.values()).map(MediaType::queryParameter).toList()
                     + " formats"
             );
         }
-        return checkNonNullMediaType(fromMediaType, request);
+    }
+
+    private static void validateIncludeCCSMetadata(boolean includeCCSMetadata, MediaType fromMediaType) {
+        if (includeCCSMetadata && fromMediaType != XContentType.JSON) {
```

Comment:
I think it's ok so long as the media type isn't text like the `instanceof` above. I expect kibana will use one of the binary ones at some point.

Reply:
OK, in the latest commit I've changed it to be like `columnar` and work with any XContent type.